libass/ass_shaper.c

```diff
@@ -639,18 +639,20 @@ static void shape_harfbuzz(ASS_Shaper *shaper, GlyphInfo *glyphs, size_t len)
     for (i = 0; i < len; i++) {
         int offset = i;
         hb_font_t *font = get_hb_font(shaper, glyphs + offset);
-        int level = glyphs[offset].shape_run_id;
-        int direction = FRIBIDI_LEVEL_IS_RTL(shaper->emblevels[offset]);
+        int run_id = glyphs[offset].shape_run_id;
+        int level = shaper->emblevels[offset];
 
         // advance in text until end of run
-        while (i < (len - 1) && level == glyphs[i+1].shape_run_id)
+        while (i < (len - 1) && run_id == glyphs[i+1].shape_run_id &&
+                level == shaper->emblevels[i+1])
```

Comment:
Shouldn't it be `[i + 1]` per libass style?

Reply:
Changed both instances to `[i + 1]`.